### PR TITLE
Made brave_unit_tests runnable for android

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/9203

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/browsing_data/counters/brave_site_settings_counter_unittest.cc [hunk 1]
```diff
@@ -46,7 +46,7 @@ class BraveSiteSettingsCounterTest : public testing::Test {
     auto* zoom_map =
         content::HostZoomMap::GetDefaultForBrowserContext(profile());
 #else
-    auto* zoom_map = nullptr;
+    content::HostZoomMap* zoom_map = nullptr;
```
Comment (samartnik) [inline browser/browsing_data/counters/brave_site_settings_counter_unittest.cc:49]: Otherwise there is error
`../../brave/browser/browsing_data/counters/brave_site_settings_counter_unittest.cc:49:11: error: variable 'zoom_map' with type 'auto *' has incompatible initializer of type 'nullptr_t'
    auto* zoom_map = nullptr;`

File: patches/chrome-android-java-src-org-chromium-chrome-browser-settings-privacy-ClearBrowsingDataFragment.java.patch [hunk 2]
```diff
@@ -1,46 +0,0 @@
-diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java
```
Comment (samartnik) [inline patches/chrome-android-java-src-org-chromium-chrome-browser-settings-privacy-ClearBrowsingDataFragment.java.patch:1]: `Clear downloads` option was removed as it didn't work as expected. If we want to get it back we will use `changeMethodOwner` next time. For now I needed to revert these changes to make it compilable after adding `//chrome/android:chrome_test_java` dependency.

File: test/BUILD.gn [hunk 3]
```diff
@@ -451,13 +459,19 @@ test("brave_unit_tests") {
...
-      "//brave/components/brave_perf_predictor/browser/named_third_party_registry_unittest.cc",
       "//brave/components/brave_perf_predictor/browser/bandwidth_linreg_unittest.cc",
       "//brave/components/brave_perf_predictor/browser/bandwidth_savings_predictor_unittest.cc",
       "//brave/components/brave_perf_predictor/browser/p3a_bandwidth_savings_permanent_state_unittest.cc",
       "//brave/components/brave_perf_predictor/browser/p3a_bandwidth_savings_tracker_unittest.cc",
     ]
 
+    if (!is_android) {
+      sources += [
+        # Fails to load `brave/components/brave_perf_predictor/resources/entities-httparchive-nostats.json` file
```
Comment (AndriusA) [inline test/BUILD.gn:470]: what's failing here? likely the loading needs fixing, but the feature wouldn't work if this actually fails to load

Comment (samartnik) [reply test/BUILD.gn:470]: We just don't have such file on android. Should it be there?

Comment (AndriusA) [reply test/BUILD.gn:470]: we should either disable `brave_perf_predictor` completely on Android, or have the file, yes

File: test/data/brave_perf_predictor/entities-httparchive-nostats.json [hunk 4]
```diff
@@ -0,0 +1 @@
...]},{"name":"AdSpruce","categories":["ad"],"domains":["adspruce.com"]},{"name":"content.ad","categories":["ad"],"domains":["content.ad"]},{"name":"Improve Digital","categories":["ad"],"domains":["360yield.com"]},{"name":"Fastest Forward","categories":["analytics"],"domains":["gaug.es"]},{"name":"RichRelevance","categories":["analytics"],"domains":["richrelevance.com"]},{"name":"ARM","categories":["analytics"],"domains":["tag4arm.com"]},{"name":"Webtrends","categories":["analytics"],"domains":["d1q62gfb8siqnm.cloudfront.net","webtrends.com","webtrendslive.com"]},{"name":"Click4Assistance","categories":["customer-success"],"domains":["click4assistance.co.uk"]},{"name":"Realytics","categories":["analytics"],"domains":["dcniko1cv0rz.cloudfront.net","realytics.net"]},{"name":"Xaxis","homepage":"https://www.xaxis.com/","categories":["ad"],"domains":["t.mookie1.com","247realmedia.com","gmads.net","odr.mookie1.com"]},{"name":"UPS i-parcel","company":"UPS","categories":["other"],"domains":["i-parcel.com"]},{"name":"Qualtrics","categories":["analytics"],"domains":["qualtrics.com"]},{"name":"Adobe Test & Target","company":"Adobe Systems","categories":["analytics"],"domains":["tt.omtrdc.net"]}]
```
Comment (bridiver) [inline test/data/brave_perf_predictor/entities-httparchive-nostats.json:1]: not sure if we want to have a copy of this cc @AndriusA

Comment (AndriusA) [reply test/data/brave_perf_predictor/entities-httparchive-nostats.json:1]: we could have the config as a component for the full build, but doesn't seem worth it since it doesn't change. There was a problem with android unit tests not finding the file (but working fine in app) - not sure if there is a more robust way of pointing unit tests to the resource file?

Comment (samartnik) [reply test/data/brave_perf_predictor/entities-httparchive-nostats.json:1]: Added `//brave/components/brave_perf_predictor/resources/entities-httparchive-nostats.json` directly as data file
